letsencrypt/DOCS.md

```diff
@@ -302,6 +302,51 @@ dns:
 ```
 Use `ovh_endpoint: ovh-ca` for north america region.
 
+### RFC2136
+You will need to set up a server with RFC2136 (Dynamic Update) support with a TKEY (to authenticate the updates).  How to do this will vary depending on the DNS server software in use.  For Bind9, you first need to first generate an authenticate key by running
+```
+# dnssec-keygen -a HMAC-SHA512 -b 512 -n HOST letsencrypt
+Kletsencrypt.+165+20675
+```
+The key file (Kletsencrypt.+165+20675.key in this example) looks like the following:
+```
+# cat Kletsencrypt.+165+20675.key
+letsencrypt. IN KEY 512 3 165 Cj2SJThIYZqZO39HIOA8dYryzsLT3CI+m43m3yfGfTMvpyYw5DXjn5da hokrwyLe3MTboGkloKIsT6DUcTSdEA==
+```
+You don't need to publish this; just copy the key data into your named.conf file:
+```
+key "letsencrypt" {
+  algorithm hmac-sha512;
+  secret "Cj2SJThIYZqZO39HIOA8dYryzsLT3CI+m43m3yfGfTMvpyYw5DXjn5da hokrwyLe3MTboGkloKIsT6DUcTSdEA==";
+};
+```
+And ensure you have an update policy in place in the zone that uses this key to enable update of the correct domain
+```
+   update-policy {
+      grant letsencrypt name _acme-challenge.hassio.dom.ain. txt;
+   };
+```
+
+For this add-on you will need to supply all the rfc2136_* options.  Note that the rfc2136_port item is required (there is no default port in the add-on) and, most importantly, the the port number must be quoted because the add-on is expecting it to be a string and not a number.  Also be sure to copy in the key so certbot can authenticate to the DNS server.
```

Comment:
Instead of document a bug, you can fix it in the schema ;)

Reply:
True, I could, but this is easier, safer, and definitely more testable :)

Reply:
@pvizeli -- coming at this with a lot of time to sit and reflect (and also understanding HA much better), another reason not to change the schema would be that it would introduce a breaking change.  If the schema says "integer" but you supply a string, wouldn't it barf on the input?  If so, that would break every current user.